docs/docs/health-checks.md

```diff
@@ -16,9 +16,123 @@ application's tasks.
+
+Marathon offers the following protocols for health checks. Declaring a protocol is optional: the default is `HTTP`.
+
+- `HTTP`
+- `HTTPS`
+- `TCP`
+- `COMMAND`
+- `MESOS_HTTP`
+- `MESOS_HTTPS`
+- `MESOS_TCP`
+
+### Marathon-level health checks
+
+Marathon-level health checks (HTTP, HTTPS, and TCP) are executed by Marathon and thus test reachability from the current Marathon leader. Marathon-level health checks have several limitations:
+
+- The API for health checks varies from one framework to the next and common functionality must be reimplemented with each new framework that requires health checks.
+
+- Marathon-level health checks create extra network traffic if the task and the scheduler are run on different nodes (this is usually the case).
+
+- Network failures between the task and the scheduler, and port mapping misconfigurations can make a healthy task look unhealthy.
+
+- If Marathon is managing a large number of tasks, performing health checks for every task can cause scheduler performance issues.
+
+These limitations may be acceptable for smaller clusters with low-scale tasks, but Marathon-level health checks should not be used on large clusters with highly scaled Marathon tasks.
+
+**Important:** Marathon-based health checks are deprecated and will be removed in a future version. See the [1.4.0 release notes](https://github.com/mesosphere/marathon/releases/tag/v1.4.0) for more information.
+
+### Mesos-level health checks
+
+Mesos-level health checks (`MESOS_HTTP`, `MESOS_HTTPS`, `MESOS_TCP`, and `COMMAND`) health checks are locally
+executed by Mesos on the agent running the corresponding task and thus test reachability from the Mesos executor. Mesos-level health checks offer the following advantages over Marathon-level health checks:
+
+- All built-in executors support native health checks via the new API and library. Custom executors can now use the same library and API to implement health checks, regardless of whether their tasks are actual processes or something else.
+
+- Mesos-level health checks are performed as close to the task as possible, so they are are not affected by networking failures.
+
+- Mesos-level health checks delegated to the agents running the tasks, so the number of tasks that can be health checked can scale horizontally with the number of agents in the cluster.
+
+#### Limitations and considerations
+
+- Mesos-level health checks consume extra resources on the agents; moreover, there is some overhead for fork-execing a process and entering the tasks’ namespaces every time a task is checked.
+
+- The health check processes share resources with the task that they check. Your application definition must account for the extra resources consumed by the health checks.
+
+- Mesos-level health checks require tasks to listen on the loopback interface in addition to whatever interface they require. If you run a service in production, you will want to make sure that the users can reach it.
```

Comment:
The loopback interface in a container is not the same as the host, unless if the container is not using network isolation.

As a user, I might be confused as to whether it needs to listen on the hosts loopback interface. Do you think this is a worth-while clarification?

> on the *container's* loopback interface